### PR TITLE
Enhance parallelise make spell number 59

### PR DESCRIPTION
**Pull request by Wartynas**
Hi @HorridTom, I have an issue when testing the optimized version of `spell_variables` (it's called `spell_variables_new`). Specifically, I observe a strange behavior with `admission_method` column when running `test_make_spell.R` script. Could you please have a look at this problem?

### CODE REVIEW

**Comment by HorridTom** [comment]
Is there a missing pipe on line 239 of the new make spell table file? Sorry limited comment as am on my phone...

**Comment by HorridTom** [comment]
Ok, cool, will have a look at the weekend... thanks!

**Comment by HorridTom** [comment]
@Wartynas I think the errors are happening because, in the dataframe saved in `tests/testthat/testdata/spell_linkage/inpatient_data_age_sex.rds`, there is no column `admission_method`, thus when `make_spell_table` is called on this data, there is no `admission_method` in the resulting `all_episodes` dataframe.

I haven't worked out why it was working when executed in the console - I can only guess that either there was an existing version of the inpatient data in the global environment that _did_ have the necessary column - perhaps created by execution of line 50 in the test file?

Let me know if this is the problem, or if we need to investigate further.

**Comment by HorridTom** [comment]
> @HorridTom, correct me if I am wrong, but the `admission_method` column is present in `tests/testthat/testdata/spell_linkage/inpatient_data_age_sex.rds`? See here:
> 
> image

Yes I do get that now... hmmm strange, yesterday I was getting something different... Will have another look later.

> 
> And I think it's `make_spell_number`, rather than `make_spell_table`, that is called on the data to generate `all_episodes` dataframe? I have an idea in mind, will let you know if it works.

Yes sorry I mis-typed, should have written `make_spell_number`. Ok yes, let me know if you get anywhere, I'll have another look later too.

**Comment by Wartynas** [comment]
@HorridTom I think I managed to find the problem, but not explain it. When the data is loaded with `readRDS` function like this : https://github.com/HorridTom/hospitalflow/blob/9ffa52a73d667455f8230726e77802fecf58b4ee/tests/testthat/test_make_spell.R#L226

the column `admission_method` gets lost. However, when the `inpatient_data_age_sex` is created manually, as in the first test in `test_make_spell.R`:

https://github.com/HorridTom/hospitalflow/blob/9ffa52a73d667455f8230726e77802fecf58b4ee/tests/testthat/test_make_spell.R#L50-L59

the `admission_method` column is present and tests run correctly. Initially I thought I could save some space by loading `.rds` files within every test, but now I am thinking of creating `inpatient_data_age_sex` and `ed_data_age_sex` dataframes manually to circumvent this peculiar behaviour at the cost of WET code.

**Comment by HorridTom** [comment]
@Wartynas I was just looking at that same line - isn't it just reading in the ed data instead of the inpatient data??

**Comment by Wartynas** [comment]
@HorridTom ... writing code in a nutshell... thanks, Tom! :)

**Comment by HorridTom** [comment]
Cool good stuff 👍

**Comment by Wartynas** [comment]
Hi @HorridTom, I think this it's ready to merge. I have updated the `make_spell_table` function in `R/spell_linkage.R`. Every test passes. Let me know if the code requires any clarification.

**Comment by Wartynas** [comment]
Distribution of the runtime of `make_spell_table` function (n=50)

716e4b29-903a-4f2b-9a40-0dc1d24f4c67

**Comment by Wartynas** [comment]
Hi @HorridTom, I think this branch is ready to be merged with `master`. All tests pass and the example report knits without errors both with `use_existing_spell_table` set to `TRUE` and `FALSE`. I have sent you an invitation for the meeting, please let me know if you'd prefer an alternative time.

File: R/make_spell_table_new.R [hunk 1]
```diff
@@ -0,0 +1,302 @@
+
```
Comment (HorridTom) [inline R/make_spell_table_new.R:1]: If functions are not needed, suggest move to a spike branch as discussed.

File: DESCRIPTION [hunk 2]
```diff
@@ -29,7 +29,11 @@ Imports:
     scales (>= 1.1.1),
     stringr (>= 1.4.0),
     tibble (>= 3.0.6),
-    tidyr (>= 1.1.2)
+    tidyr (>= 1.1.2),
+    future (>= 1.24.0),
+    furrr (>= 0.2.0),
```
Comment (HorridTom) [inline DESCRIPTION:34]: Ensure all packages added to DESCRIPTION are necessary for the final functionality to be merged

File: experiments/parallel_make_spell_number.Rmd [hunk 3]
```diff
@@ -0,0 +1,201 @@
+---
```
Comment (HorridTom) [inline experiments/parallel_make_spell_number.Rmd:1]: As above - spike

**Review by HorridTom — CHANGES_REQUESTED** [review]
See comments and just check if those "Joining by" warnings can be removed by fully specifying the join column each time...

**Comment by Wartynas** [comment]
@HorridTom, the issues should be fixed now (no messages are printed during left_join, removed unnecessary dependencies and experimentation files). Please have a look and let me know if there is something else

**Pull request merged**